lib/matplotlib/colorbar.py

```diff
@@ -222,102 +224,69 @@ def _set_ticks_on_axis_warn(*args, **kw):
+    over/under colors.
+
+    Users should not normally instantiate this class, but it is the class
+    returned by ``cbar = fig.colorbar(im); cax = cbar.ax``.
     """
-
-    def __init__(self, colorbar):
+    def __init__(self, parent, userax=False):
         """
-        This ticker needs to know the *colorbar* so that it can access
-        its *vmin* and *vmax*.  Otherwise it is the same as
-        `~.ticker.AutoLocator`.
-        """
-
-        self._colorbar = colorbar
-        nbins = 'auto'
-        steps = [1, 2, 2.5, 5, 10]
-        super().__init__(nbins=nbins, steps=steps)
-
-    def tick_values(self, vmin, vmax):
-        # flip if needed:
-        if vmin > vmax:
-            vmin, vmax = vmax, vmin
-        vmin = max(vmin, self._colorbar.norm.vmin)
-        vmax = min(vmax, self._colorbar.norm.vmax)
-        ticks = super().tick_values(vmin, vmax)
-        rtol = (vmax - vmin) * 1e-10
-        return ticks[(ticks >= vmin - rtol) & (ticks <= vmax + rtol)]
-
-
-class _ColorbarAutoMinorLocator(ticker.AutoMinorLocator):
-    """
-    AutoMinorLocator for Colorbar
-
-    This locator is just a `.AutoMinorLocator` except the min and max are
-    clipped by the norm's min and max (i.e. vmin/vmax from the
-    image/pcolor/contour object).  This is necessary so that the minorticks
-    don't extrude into the "extend regions".
-    """
-
-    def __init__(self, colorbar, n=None):
-        """
-        This ticker needs to know the *colorbar* so that it can access
-        its *vmin* and *vmax*.
+        Parameters
+        ----------
+        parent : Axes
+            Axes that specifies the position of the colorbar.
+        userax : boolean
+            True if the user passed `.Figure.colorbar` the axes manually.
         """
-        self._colorbar = colorbar
-        self.ndivs = n
-        super().__init__(n=None)
-
-    def __call__(self):
-        vmin = self._colorbar.norm.vmin
-        vmax = self._colorbar.norm.vmax
-        ticks = super().__call__()
-        rtol = (vmax - vmin) * 1e-10
-        return ticks[(ticks >= vmin - rtol) & (ticks <= vmax + rtol)]
-
-
-class _ColorbarLogLocator(ticker.LogLocator):
-    """
-    LogLocator for Colorbarbar
-
-    This locator is just a `.LogLocator` except the min and max are
-    clipped by the norm's min and max (i.e. vmin/vmax from the
-    image/pcolor/contour object).  This is necessary so ticks don't
-    extrude into the "extend regions".
 
-    """
-    def __init__(self, colorbar, *args, **kwargs):
+        if userax:
+            # copy position:
+            fig = parent.figure
+            parent_ax = fig.add_axes(parent.get_position())
+            # copy the locator if one exists:
+            parent_ax._axes_locator = parent._axes_locator
+            # remove the parent from the figure...
+            parent.remove()
+        else:
+            parent_ax = parent
+
+        inner_ax = parent_ax.inset_axes([0, 0, 1, 1])
```

Comment:
You use `inner` and `parent` throughout, I think it would flow better as one of the pairs (`inner`, `outer`) or (`child`, `parent`) instead. I would vote in favor of inner/outer as that seems to signify this inset axes relationship a little better, but don't have a real strong preference at all.

Reply:
Sure, did inner/outer....